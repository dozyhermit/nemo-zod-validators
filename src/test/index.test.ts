import { SafeParseReturnType, z } from 'zod';
import { validateGeneric } from '../';

const schema = z.object({
  Number: z.number({
    invalid_type_error: 'Invalid Number type',
  }),
  String: z.string({
    invalid_type_error: 'Invalid String type',
  }),
  AdvancedString: z
    .string({
      invalid_type_error: 'Invalid AdvancedString type',
    })
    .regex(/^[0-9]+$/, 'Invalid AdvancedString regex'),
});

type SchemaType = z.infer<typeof schema>;

// Handler required to avoid testing with Next.js responses
const errorHandler = ({
  error,
}: SafeParseReturnType<Record<string, any>, SchemaType>) =>
  error?.flatten()?.fieldErrors;

describe('validateGeneric', () => {
  test('Triggers validation error with incorrectly supplied data', () => {
    expect(
      validateGeneric<SchemaType>({
        // AdvancedString must be a number string
        data: { Number: 123, String: 'Hello', AdvancedString: 'World' },
        schema,
        errorHandler,
      })
    ).toMatchSnapshot();
  });

  test('Passes validation with correctly supplied data', () => {
    expect(
      validateGeneric<SchemaType>({
        data: { Number: 123, String: 'Hello', AdvancedString: '123456' },
        schema,
        errorHandler,
      })
    ).toMatchSnapshot();
  });
});
