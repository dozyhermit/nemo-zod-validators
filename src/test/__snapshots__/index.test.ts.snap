// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateGeneric Passes validation with correctly supplied data 1`] = `
NextResponse {
  Symbol(state): {
    "aborted": false,
    "cacheState": "",
    "headersList": HeadersList {
      "cookies": null,
      Symbol(headers map): Map {
        "x-middleware-next" => {
          "name": "x-middleware-next",
          "value": "1",
        },
      },
      Symbol(headers map sorted): null,
    },
    "rangeRequested": false,
    "requestIncludesCredentials": false,
    "status": 200,
    "statusText": "",
    "timingAllowPassed": false,
    "timingInfo": null,
    "type": "default",
    "urlList": [],
  },
  Symbol(headers): Headers {},
  Symbol(internal response): {
    "cookies": bound ResponseCookies {
      "_headers": Headers {},
      "_parsed": Map {},
    },
    "url": undefined,
  },
}
`;

exports[`validateGeneric Triggers validation error with incorrectly supplied data 1`] = `
{
  "AdvancedString": [
    "Invalid AdvancedString regex",
  ],
}
`;
